pkg/csi/livenessprobe/livenessprobe.go

```diff
@@ -84,26 +86,32 @@ func (s *Server) probeRequest(w http.ResponseWriter, r *http.Request) {
 
 	ready, err := rpc.Probe(ctx, conn)
 	if err != nil {
-		w.WriteHeader(http.StatusInternalServerError)
-		w.Write([]byte(err.Error()))
+		writeResponse(w, http.StatusInternalServerError, err.Error())
 		log.Error(err, "health check failed")
 
 		return
 	}
 
 	if !ready {
-		w.WriteHeader(http.StatusInternalServerError)
-		w.Write([]byte("driver is not ready"))
+		writeResponse(w, http.StatusInternalServerError, "driver is not ready")
 		log.Error(nil, "driver is not ready")
 
 		return
 	}
+	writeResponse(w, http.StatusOK, "ok")
 
-	w.WriteHeader(http.StatusOK)
-	w.Write([]byte(`ok`))
 	log.Debug("health check succeeded")
 }
 
+func writeResponse(w http.ResponseWriter, statusCode int, message string) {
+	w.WriteHeader(statusCode)
+
+	_, err := w.Write([]byte(message))
```

Comment:
could use [`http.Error`](https://pkg.go.dev/net/http#Error) instead of this function

Reply:
http.Error just wraps setting the error code and writing the message + adding some headers

Reply:
honestly, I would not want to write `http.Error(w, "ok",  http.StatusOK)` 😅 

(can != should)